### PR TITLE
Supplying authorization to get slide notes for ottrpal

### PR DESCRIPTION
**Pull request by cansavvy**
#### What changes are being implemented in this Pull Request?
This was something we pair coded on. Basically if credentials aren't supplied we want to use a dummy credential to grab slides that are public anyway. 

This code should do that with some encryption steps as well. 

This is a PR based off the conversation here: https://github.com/jhudsl/ottrpal/pull/116#issuecomment-1661231230

### CODE REVIEW

File: R/auth.R [hunk 1]
```diff
@@ -1,74 +1,127 @@
...
+
+  if (is.null(access_token) | is.null(refresh_token)) {
+    decrypted <- openssl::aes_cbc_decrypt(
+      readRDS(encrypt_creds_user_path()),
+      key = readRDS(key_encrypt_creds_path())
+    )
+  }
+
+  credentials <- list(
+    access_token =  unserialize(decrypted)[[1]]$access_token,
```
Comment (howardbaek) [inline R/auth.R:89]: If user provides the `access_token` and `refresh_token` as arguments, then this line won't be able to find `decrypted` object.

File: R/auth.R [hunk 2]
```diff
@@ -1,74 +1,127 @@
...
+    expires_in = 3599L,
+    refresh_token = unserialize(decrypted)[[1]]$refresh_token,
```
Comment (howardbaek) [inline R/auth.R:91]: Same comment as above.

File: R/auth.R (hunk 1 above)
Comment (cansavvy) [reply R/auth.R:89]: Oh right! We need to have this defined as `access_token` and `refresh_token` above in the if statement

**Review by cansavvy — COMMENTED** [review]
Does this comment make sense? @howardbaek Do you know how to fix this? If not I can send suggestions.

**Comment by howardbaek** [comment]
> 

It seems like `auth_from_secret()` is a function that uses the "secrets" that we provided to the ottrpal R package to obtain a token. If so, should we just get rid of all the function arguments (`access_token` and `refresh_token`) so that when we run `auth_from_secret()`, we run this chunk no matter what:

```
decrypted <- openssl::aes_cbc_decrypt(
      readRDS(encrypt_creds_user_path()),
      key = readRDS(key_encrypt_creds_path())
)
```

and we create `decrypted`. When I supply a value to `access_token` and `refresh_token`, then the above code chunk doesn't run since its inside an if statement, and the following code chunk throws an error since it doesn't know what `decrypted` is:

```
  credentials <- list(
    access_token =  unserialize(decrypted)[[1]]$access_token,
    expires_in = 3599L,
    refresh_token = unserialize(decrypted)[[1]]$refresh_token,
    scope = scopes_list,
    token_type = "Bearer"
  )
```

File: R/auth.R [hunk 3]
```diff
@@ -1,74 +1,121 @@
...
+#'   token$credentials$refresh_token
+#' )
+#' }
+#'
+auth_from_secret <- function(access_token = NULL, refresh_token = NULL) {
+  if (is.null(access_token) | is.null(refresh_token)) {
+    decrypted <- openssl::aes_cbc_decrypt(
+      readRDS(encrypt_creds_user_path()),
+      key = readRDS(key_encrypt_creds_path())
+    )
```
Comment (cansavvy) [inline R/auth.R:84]:
```suggestion
    )
access_token <- unserialize(decrypted)[[1]]$access_token
refresh_token <- unserialize(decrypted)[[1]]$refresh_token
```
I'm doing this on my phone but this is what I'm suggesting you do. Bring the phrases from below up here (not sure if I typed these right) and declare the variables. Then below just specify `access_token = access_token` and same for refresh_token.

If this doesn't make sense, let me know and I'll send a commit later.

**Review by cansavvy — COMMENTED** [review]
@howardbaek this is what I was trying to describe to fix this. Let me know if this makes sense.

**Comment by howardbaek** [comment]
@cansavvy  Looks good. Merged.

**Pull request merged**